core/src/main/java/hudson/Util.java

```diff
@@ -1186,6 +1179,22 @@ public static void createSymlink(@Nonnull File baseDir, @Nonnull String targetPa
             Path path = fileToPath(new File(baseDir, symlinkPath));
             Path target = Paths.get(targetPath, MemoryReductionUtil.EMPTY_STRING_ARRAY);
 
+            try {
+                File symlink = File.createTempFile("tmp-"+symlinkPath, null, baseDir);
```

Comment:
So this is intending to replace the code in #1201 ? Why not there?

Reply:
This code is atomic, it should either work with a single attempt, or fail. If it fails, the older code can be used, including its multiple attempts.

Reply:
If this code is atomic and thus more appropriate, why maintain the older one?

Reply:
I'm assuming there are cases where it won't work. I'm looking for an incremental improvement, not a boil the ocean one.